src/app/public/src/modules/code-block/code-block.component.html

```diff
@@ -1,6 +1,14 @@
 <div class="stache-code-block">
-  <div #codeFromContent class="stache-code-input"><ng-content></ng-content></div>
-  <pre class="stache-code-output"><code
-    [ngClass]="getClassName()"
-    [innerHTML]="output"></code></pre>
-</div>
\ No newline at end of file
+  <div #codeFromContent class="stache-code-input">
+    <ng-content></ng-content>
+  </div>
+  <div>
+    <pre #stacheCodeOutput class="stache-code-output"><code
+        [ngClass]="getClassName()"
+        [innerHTML]="output"></code></pre>
+  </div>
+  <stache-copy-to-clipboard
+    [copyTarget]="stacheCodeOutput"
+    buttonText="{{ 'stache_copy_to_clipboard' | skyAppResources }}">
```

Comment:
Instead of setting this here, we should probably make it the default within the clipboard component itself (I noticed that the clipboard doesn't set a default for the button text).

**clipboard.component.ts**

```
@Input()
public set buttonText(value: string) {
  this._buttonText = value;
}

public get buttonText(): string {
  return this._buttonText || this.defaultButtonText;
}

private _buttonText: string;
private defaultButtonText: string;

constructor(
  private resourcesService: SkyAppResourcesService
) { }

public ngOnInit(): void {
  this.resourcesService.getString('stache_copy_to_clipboard')
    .subscribe((value: string) => {
      this.defaultButtonText = value;
    });
}
```

Reply:
I debated with this.  The reason it's not setting the default text within with component is to avoid forcing the button to contain text at all.  If the user just wanted the icon (for example next to a form Input) they don't have to supply text.